src/plugins/intel_gna/src/transformations/ts_split.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/ts_split.hpp"
+
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "backend/gna_limitations.hpp"
+#include "common/graph_utils.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset12.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/rt_info/transpose_sinking_attr.hpp"
+#include "transformations/utils/transformation_helper.hpp"
+
+using namespace ov;
+using namespace ov::opset12;
+using namespace ov::pass::pattern;
+using namespace ov::intel_gna::pass;
+using namespace ov::intel_gna::pass::helper;
+using namespace ov::intel_gna::limitations;
+using namespace ov::intel_gna::graph_utils;
+
+namespace {
+bool is_split_sinked(const Output<Node>& output) {
+    auto split_node = output.get_node_shared_ptr();
+    for (size_t output_idx = 0; output_idx < split_node->get_output_size(); ++output_idx) {
+        for (auto& input : split_node->get_output_target_inputs(output_idx)) {
+            auto transpose = ov::as_type_ptr<Transpose>(input.get_node()->shared_from_this());
+            if (transpose && !Limitations::is_transpose_supported(transpose))
+                return true;
+        }
+    }
+    return false;
+}
+
+std::vector<size_t> create_gather_indices(const ov::Shape& input_shape, const ov::Shape& order) {
+    if (input_shape.size() < 2 || input_shape.size() > 4) {
+        THROW_GNA_EXCEPTION << "Unsupported shape size: " << input_shape.size();
+    }
+
+    ov::Shape input_shape_4d = input_shape;
+    ov::Shape order_4d = order;
+    // Just to simplify the code we transform all shapes to 4d by adding 1 dimensions at the end
+    while (input_shape_4d.size() < 4) {
+        input_shape_4d.push_back(1);
+        order_4d.push_back(order_4d.size());
+    }
+    ov::Shape output_shape_4d = transpose_shape(input_shape_4d, order_4d);
+
+    // common case when shape is 4d
+    std::vector<size_t> xyz_4d = {input_shape_4d[3] * input_shape_4d[2] * input_shape_4d[1],
+                                  input_shape_4d[3] * input_shape_4d[2],
+                                  input_shape_4d[3],
+                                  1};
+
+    std::vector<size_t> xyz = transpose_shape(xyz_4d, order_4d);
+    std::vector<size_t> gather_order;
+
+    for (size_t n = 0; n < output_shape_4d[0]; ++n) {
```

Comment:
How many times are we going to duplicate this code? please use make_gather_indices_from_transpose_axes